node-server/app.js

```diff
@@ -0,0 +1,429 @@
+    });
+
+    this.name = 'TaskNotFoundError';
+}
+
+util.inherits(TaskNotFoundError, restify.RestError);
+
+/**
+ * Our Server
+ */
+
+
+var server = restify.createServer({
+    name: "Windows Azure Active Directroy TODO Server",
+    version: "2.0.1"
+});
+
+// Ensure we don't drop data on uploads
+server.pre(restify.pre.pause());
+
+// Clean up sloppy paths like //todo//////1//
+server.pre(restify.pre.sanitizePath());
+
+// Handles annoying user agents (curl)
+server.pre(restify.pre.userAgentConnection());
+
+// Set a per request bunyan logger (with requestid filled in)
+server.use(restify.requestLogger());
+
+// Allow 5 requests/second by IP, and burst to 10
+server.use(restify.throttle({
+    burst: 10,
+    rate: 5,
+    ip: true,
+}));
+
+// Use the common stuff you probably want
+server.use(restify.acceptParser(server.acceptable));
+server.use(restify.dateParser());
+server.use(restify.queryParser());
+server.use(restify.gzipResponse());
+server.use(restify.bodyParser({
+    mapParams: true
+})); // Allows for JSON mapping to REST
+server.use(restify.authorizationParser()); // Looks for authorization headers
+
+// Let's start using Passport.js
+
+server.use(passport.initialize()); // Starts passport
+server.use(passport.session()); // Provides session support
+
+/**
+/*
+/* Calling the OIDCBearerStrategy and managing users
+/*
+/* Passport pattern provides the need to manage users and info tokens
+/* with a FindorCreate() method that must be provided by the implementor.
+/* Here we just autoregister any user and implement a FindById().
+/* You'll want to do something smarter.
+**/
+
+var findById = function(id, fn) {
+    for (var i = 0, len = users.length; i < len; i++) {
+        var user = users[i];
+        if (user.oid === id) {
+            log.info('Found user: ', user);
+            return fn(null, user);
+        }
+    }
+    return fn(null, null);
+};
+
+
+var bearerStrategy = new OIDCBearerStrategy(options,
+    function(token, done) {
+        log.info('verifying the user');
+        log.info(token, 'was the token retreived');
+        findById(token.oid, function(err, user) {
+            if (err) {
+                return done(err);
+            }
+            if (!user) {
+                // "Auto-registration"
+                log.info('User was added automatically as they were new. Their oid is: ', token.oid);
+                users.push(token);
+                owner = token.oid;
+                return done(null, token);
+            }
+            owner = token.oid;
+            return done(null, user, token);
+        });
+    }
+);
+
+passport.use(bearerStrategy);
+
+/// Now the real handlers. Here we just CRUD
+
+/**
+/*
+/* Each of these handlers are protected by our OIDCBearerStrategy by invoking 'oidc-bearer'
+/* in the pasport.authenticate() method. We set 'session: false' as REST is stateless and
+/* we don't need to maintain session state. You can experiement removing API protection
```

Comment:
if so, why do we have 
var users = []   ->  // array to hold logged in users and the current logged in user (owner)
?